graphql_compiler/compiler/emit_sql.py

```diff
@@ -103,6 +108,83 @@ def _find_folded_outputs(ir):
+    the XML PATH from having extraneous XML tags based on the alias given to the column.
+    """
+
+    pass
+
+
+@compiles(_CompileLabel, 'default')
+def compile_xmlpath(element, compiler, **kw):
+    """Suppress labeling when compiling XML PATH subqueries, otherwise compile as usual.
+
+    This method is needed because if XML PATH column is selected with an alias then each
+    entry in the resulting array is wrapped in XML tags bearing that alias.
+    e.g. `SELECT (SELECT ... FOR XML PATH ('')) AS blah ...` results in a string of the form
+    <blah>element1</blah><blah>element2</blah>. Since SQLAlchemy adds labels to all Selectables, we
+    must to suppress that labeling.
+    """
+    if isinstance(element.element, XMLPathBinaryExpression):
+        # this indicates that no label should be inserted for this element
+        kw.update(
+            within_columns_clause=False  # label always gets added if within_columns_clause is True
+        )
+    return compiler.visit_label(element, **kw)
+
+
+def get_xml_path_clause(output_column, where):
+    """Produce an MSSQL-style XML PATH-based aggregation subquery.
+
+    Args:
+        output_column: SQLAlchemy Column, value being aggregated with XML PATH
+        where: SQLAlchemy BinaryExpression, predicate used to filter
+
+    Returns:
+        An SQLAlchemy Selectable corresponding to an XML PATH subquery
+        which aggregates the values of the output_column from the output vertex.
+        Empty arrays are represented as '', null values are represented as '~', array
+        entries are delimited using '|'.
+
+        All occurrences of '^', '~', and '|' in the original string values are
+        replaced with '^e', '^n', and '^d', resp.
+
+        Undoing the encoding above as well as the XML reference entity encoding performed
+        by the XML PATH statement is deferred to post-processing when the list is retrieved
+        from the string representation produced by the subquery.
+
+        Post-processing must split on '|', convert '~' to None, and both the encoding above
+        and the XML reference entity encoding auto performed by XML PATH. In particular, undo
+        '^d' -> '|', '^e' -> '^', '^n' -> '~', '&amp;' -> '&', '&gt;' -> '>', '&lt;' -> '<',
+        '&0xHEX;' -> u'0xHEX'
+
+        Joining from the preceding vertex to the correct output vertex
+        is performed with the WHERE clause.
+    """
+    # delimit elements in the array using '|'
+    xml_column = (u'|' + func.COALESCE(  # denote null values with '~'
+        func.REPLACE(  # replace all occurrences of '|' in the original with '|d'
+            func.REPLACE(  # replace all occurrences of '~' in the original with '|n'
+                func.REPLACE(  # replace all occurrences of '^' in the original with '|e'
+                    output_column, '^', '^e'
+                ), '~', '^n'
+            ), '|', '^d'
+        ), '~'
```

Comment:
Can you see if using literals here instead of strings produces literals in the query?